deep_quoridor/src/train_alphazero.py

```diff
@@ -151,6 +156,40 @@ def train_alphazero(
         wandb_train_plugin.end_arena(None, [])
 
 
+def dedup_replay_buffer_items(existing_items: list, new_items: list, dedup_mode: str) -> list:
```

Comment:
Will this method of de-duplication result in prioritizing the oldest examples of each state? If it does, those will still have the values that they got from self-play using outdated versions of the model.

For example, lets say you've got 4 examples of a state in the replay buffer from previous epochs. Then in the next epoch you get 2 more. Those two are ignored because of de-duplication, even though their values are are from the most up-to date version of the model. Would it make more sense to replace the oldest examples of a state when you get new examples (but without reaching the next power of 2)

Reply:
Great idea! The implementation for that is tricky, so I'll work on it on a separate PR.